finality-aleph/src/metrics.rs

```diff
@@ -81,37 +97,43 @@ impl Inner {
             hash,
             checkpoint_time
         );
-
-        self.starts.entry(checkpoint_type).and_modify(|starts| {
-            starts.put(hash, checkpoint_time);
-        });
-
-        if let Some(prev_checkpoint_type) = self.prev.get(&checkpoint_type) {
-            if let Some(start) = self
-                .starts
-                .get_mut(prev_checkpoint_type)
-                .expect("All checkpoint types were initialized")
-                .get(&hash)
-            {
-                let duration = match checkpoint_time.checked_duration_since(*start) {
-                    Some(duration) => duration,
-                    None => {
-                        warn!(
-                            target: LOG_TARGET,
-                            "Earlier metrics time {:?} is later that current one \
-                        {:?}. Checkpoint type {:?}, block: {:?}",
-                            *start,
-                            checkpoint_time,
-                            checkpoint_type,
-                            hash
-                        );
-                        Duration::new(0, 0)
-                    }
-                };
-                self.gauges
-                    .get(&checkpoint_type)
+        if let BlockMetrics::Prometheus {
+            prev,
+            gauges,
+            starts,
+        } = self
+        {
```

Comment:
nit:
```suggestion
let (prev, gauges, start) = match self {
        BlockMetrics::Prometheus {
            prev,
            gauges,
            starts,
        } => (...),
        _ => return,
};
```

Reply:
Here we replace `if let` with `match`, and later `match` with `if let` :smiley:? Nevertheless, I think struct deconstruction with match + early return when Noop is both prettier and has smaller nesting